src/sos_matlab/converter.py

```diff
@@ -33,13 +33,16 @@ def convert(self, script_file, notebook_file, args=None, unknown_args=None):
             metainfo = {}
             if args.use_sos:
                 metainfo = {'kernel': 'SoS'}
-            cells = [
-                new_code_cell(
-                    # remove any trailing blank lines...
-                    source=src.read(),
-                    execution_count=1,
-                    metadata=metainfo)
-            ]
+            cells = []
+            lines = re.split(r'\n\n+', src.read())
+            count = 1
+            for line in lines: 
+                cells.append(
+                    new_code_cell(
+                    source=line, 
+                    execution_count=count, 
+                    metadata=metainfo))
+                count = count + 1
```

Comment:
1. Can you handle cases with optional spaces between new lines?
2. Can I challenge you to use list comprehension instead of a loop? To handle `count` with list comprehension you will also need to use `enumerate` function.